lib/ui/profitloss/RecordAmount.dart

```diff
@@ -0,0 +1,295 @@
+  String buttonTitle;
+  bool isFilled;
+  Function onTap;
+  RecordType type;
+  bool isEditable;
+  String description;
+
+  RecordAmountViewModel(
+      {this.loadingStatus,
+      this.actions,
+      this.amount: "00",
+      this.buttonTitle,
+      this.isFilled: false,
+      this.onTap,
+      this.type,
+      this.isEditable: true});
+}
+
+class RecordAmountStyle {
+  final EdgeInsets buttonEdgePadding;
+
+  final EdgeInsets appBarLeftMargin;
+  final EdgeInsets appBarRightMargin;
+
+  final double headerLineSpace;
+  final double appBarElevation;
+  final double appBarIconHeight;
+  final double appBarIconWidth;
+
+  const RecordAmountStyle(
+      {this.buttonEdgePadding,
+      this.appBarLeftMargin,
+      this.appBarRightMargin,
+      this.headerLineSpace,
+      this.appBarElevation,
+      this.appBarIconHeight,
+      this.appBarIconWidth});
+
+  RecordAmountStyle copyWit(
+      {EdgeInsets buttonEdgePadding,
+      EdgeInsets appBarLeftMargin,
+      EdgeInsets appBarRightMargin,
+      double headerLineSpace,
+      double appBarElevation,
+      double appBarIconHeight,
+      double appBarIconWidth}) {
+    return RecordAmountStyle(
+        buttonEdgePadding: buttonEdgePadding ?? this.buttonEdgePadding,
+        appBarLeftMargin: appBarLeftMargin ?? this.appBarLeftMargin,
+        appBarRightMargin: appBarRightMargin ?? this.appBarRightMargin,
+        headerLineSpace: headerLineSpace ?? this.headerLineSpace,
+        appBarElevation: appBarElevation ?? this.appBarElevation,
+        appBarIconHeight: appBarIconHeight ?? this.appBarIconHeight,
+        appBarIconWidth: appBarIconWidth ?? this.appBarIconWidth);
+  }
+}
+
+class _DefaultStyle extends RecordAmountStyle {
+  final EdgeInsets buttonEdgePadding = const EdgeInsets.all(32.0);
+
+  final EdgeInsets appBarLeftMargin = const EdgeInsets.only(left: 31);
+  final EdgeInsets appBarRightMargin = const EdgeInsets.only(right: 0);
+
+  final double headerLineSpace = 12.5;
+
+  final double appBarElevation = 0;
+  final double appBarIconHeight = 20;
+  final double appBarIconWidth = 20.5;
+
+  const _DefaultStyle(
+      {EdgeInsets buttonEdgePadding,
+      EdgeInsets appBarLeftMargin,
+      EdgeInsets appBarRightMargin,
+      double headerLineSpace,
+      double appBarElevation,
+      double appBarIconHeight,
+      double appBarIconWidth});
+}
+
+const RecordAmountStyle _defaultStyle = const _DefaultStyle();
+
+class RecordAmount extends StatefulWidget {
+  final RecordAmountViewModel _viewModel;
+  final RecordAmountStyle _style;
+
+  RecordAmount(
+      {Key key,
+      RecordAmountViewModel viewModel,
+      RecordAmountStyle style = _defaultStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  RecordAmountState createState() => RecordAmountState();
+}
+
+class RecordAmountState extends State<RecordAmount> {
+  String selectedCrop;
+  bool amoundIsFilled = false;
+  bool cropIsFilled = false;
```

Comment:
should all these be public?

Reply:
Two of them should be public, because we access them through RecordAmountListItem (selectedCrop, cropIsFilled) the rest can be private.